File: backend/LearningDashboard.Api/Entities/ProjectTask.cs
using System.ComponentModel.DataAnnotations;

namespace LearningDashboard.Api.Entities;

public class ProjectTask
{
    public int Id { get; set; }

    [Required]
    [MaxLength(200)]
    public string Title { get; set; } = string.Empty;

    [MaxLength(2000)]
    public string? Description { get; set; }

    [MaxLength(100)]
    public string? Category { get; set; }

    public TaskPriority Priority { get; set; } = TaskPriority.Medium;

    public TaskStatus Status { get; set; } = TaskStatus.Planned;

    public int OwnerId { get; set; }

    public User Owner { get; set; } = null!;

    public DateTime? DueDate { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }
}

File: frontend/src/types/projectTask.ts
import type { TaskPriority, TaskStatus } from './enums'

export interface ProjectTask {
  id: number
  title: string
  description: string | null
  category: string | null
  priority: TaskPriority
  status: TaskStatus
  ownerId: number
  ownerName: string
  dueDate: string | null
  createdAt: string
  updatedAt: string
}

export interface CreateProjectTaskRequest {
  title: string
  description?: string | null
  category?: string | null
  priority?: TaskPriority
  status?: TaskStatus
  ownerId: number
  dueDate?: string | null
}

export interface UpdateProjectTaskRequest {
  title: string
  description?: string | null
  category?: string | null
  priority?: TaskPriority
  status?: TaskStatus
  ownerId: number
  dueDate?: string | null
}

export interface UpdateTaskStatusRequest {
  status: TaskStatus
}
